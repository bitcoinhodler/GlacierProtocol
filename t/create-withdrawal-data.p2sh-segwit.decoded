{
  "txid": "a339cc05314f1b7f601ecc0ef11ed98de1e32b8745633ee2b14777e1d7b29cda",
  "hash": "7797ea0457459dc6f91976decee4846eb2b5353915e7159673d52905fd23d9d7",
  "version": 2,
  "size": 440,
  "vsize": 224,
  "weight": 896,
  "locktime": 0,
  "vin": [
    {
      "txid": "967df4155ffdc72e28b10f4a5082f4b751d48287372bcbd9d1df2ce22bea02a0",
      "vout": 1,
      "scriptSig": {
        "asm": "002065c863c750331cc03efb4daca217e7925cff87f5e9ca9927564234be0d6278e8",
        "hex": "22002065c863c750331cc03efb4daca217e7925cff87f5e9ca9927564234be0d6278e8"
      },
      "txinwitness": [
        "",
        "304402204a3da582824adf22c31487bd5124b9b8f95b0dda0d276b8f4e4ff38278ebd90e02204f24ba08eaa38a996bd40e6d56717e6314dffd949d234e0e2cca44b8b97863ff01",
        "30440220431d849d362c6c6b3e93f2f33acae32ce92ba75b566baa43dec074e6d32531f7022043e673f33faa05b580e715b11d526390cad22d4ac5414960ce50b10b726cf6bb01",
        "522103d14ddcfb6817f5579695bbb3eb3e185887bf2942b031e6f716343b8fe7e9e8e221028fcd46f8614b2cbf318096968242a1e22bcfb6d8f2b6dc939c8c27c347b2937b210315acb550120f4cdcb460d5c49080ab508ca4ccd8dedecac22feeac8cd017d4c121022b063ee2f22f9e1982c140fe778672c683111a796dcccdbd47bb65e3d608a98354ae"
      ],
      "sequence": 4294967295
    }
  ],
  "vout": [
    {
      "value": 0.06997760,
      "n": 0,
      "scriptPubKey": {
        "asm": "OP_HASH160 ad50fb2489353008e75b0193011ff1f4b76f798c OP_EQUAL",
        "hex": "a914ad50fb2489353008e75b0193011ff1f4b76f798c87",
        "reqSigs": 1,
        "type": "scripthash",
        "addresses": [
          "2N93du8YobdgsHyu3qgBvSyhGUT52utMNeA"
        ]
      }
    },
    {
      "value": 0.03000000,
      "n": 1,
      "scriptPubKey": {
        "asm": "OP_DUP OP_HASH160 c3f4800772d58f5429dcb550b1a6a53580175bc1 OP_EQUALVERIFY OP_CHECKSIG",
        "hex": "76a914c3f4800772d58f5429dcb550b1a6a53580175bc188ac",
        "reqSigs": 1,
        "type": "pubkeyhash",
        "addresses": [
          "myP4xdJNwAW9iMakvCjnozg814ewgn8apd"
        ]
      }
    }
  ]
}
