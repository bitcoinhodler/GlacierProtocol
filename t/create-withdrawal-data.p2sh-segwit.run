#!/bin/bash
set -e

$GLACIERSCRIPT --regtest=$1 create-withdrawal-data << INPUT
y
y
y
y
y
y
2N93du8YobdgsHyu3qgBvSyhGUT52utMNeA
522103d14ddcfb6817f5579695bbb3eb3e185887bf2942b031e6f716343b8fe7e9e8e221028fcd46f8614b2cbf318096968242a1e22bcfb6d8f2b6dc939c8c27c347b2937b210315acb550120f4cdcb460d5c49080ab508ca4ccd8dedecac22feeac8cd017d4c121022b063ee2f22f9e1982c140fe778672c683111a796dcccdbd47bb65e3d608a98354ae
myP4xdJNwAW9iMakvCjnozg814ewgn8apd
1
020000000129bee240ee7e666d3b177881b25de9f8ad33766ae6822026ec2344084f404c40010000006a473044022074824c797034fe4499bdf574541dee5058b8ed77053ad7df00d7d8855159fea402200ecdc3ef50962423d8906ae7245691a8dea772bf66794efa6eecd4a3e9cf1c37012103f4d062a6af014545360c97bc459b10becd32ce07195902fb73f767ed071f4d1fffffffff02d3f55b05000000001976a914e523fde08b6e1fb6db6eceb3ccee7ae2a5dfe83288ac809698000000000017a914ad50fb2489353008e75b0193011ff1f4b76f798c8700000000
2
cQCrT9Ncs9729ao7jbmAWrD9z7tF64s2yKzmD6nkiLAi9sXVZWAn
cP65UeSDZPiTLB6CBwasWv9oJYEjRgQXhswfwcT9HscEKDcEbgy4
10
y
0.03000000
y
INPUT
