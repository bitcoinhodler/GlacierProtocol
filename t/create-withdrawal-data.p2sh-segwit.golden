Are you running this on a computer WITHOUT a network connection of any kind? (y/n)?Have the wireless cards in this computer been physically removed? (y/n)?Are you running on battery power? (y/n)?Are you running on an operating system booted from a USB drive? (y/n)?Is your screen hidden from view of windows, cameras, and other people? (y/n)?Are smartphones and all other nearby devices turned off and in a Faraday bag? (y/n)?
You will need to enter several pieces of information to create a withdrawal transaction.


*** PLEASE BE SURE TO ENTER THE CORRECT DESTINATION ADDRESS ***


Source cold storage address: 
Redemption script for source cold storage address: 
Destination address: 
How many unspent transactions will you be using for this withdrawal? 
Please paste raw transaction #1 (hexadecimal format) with unspent outputs at the source address
OR
input a filename located in the current directory which contains the raw transaction data
(If the transaction data is over ~4000 characters long, you _must_ use a file.):

Transaction data found for source address.
TOTAL unspent amount for this raw transaction: 0.10000000 BTC

How many private keys will you be signing this transaction with? 
#: Key #1: Key #2: 
Enter fee rate.
Satoshis per vbyte: 
Based on the provided rate, the fee will be 0.00002240 bitcoin.
Confirm? (y/n): 
Please enter the decimal amount (in bitcoin) to withdraw to the destination address.

Example: For 2.3 bitcoins, enter "2.3".

After a fee of 0.00002240, you have 0.09997760 bitcoins available to withdraw.

*** Technical note for experienced Bitcoin users:  If the withdrawal amount & fee are cumulatively less than the total amount of the unspent transactions, the remainder will be sent back to the same cold storage address as change. ***

Amount to send to myP4xdJNwAW9iMakvCjnozg814ewgn8apd (leave blank to withdraw all funds stored in these unspent transactions): 0.06997760 being returned to cold storage address address 2N93du8YobdgsHyu3qgBvSyhGUT52utMNeA.

Is this data correct?
*** WARNING: Incorrect data may lead to loss of funds ***

0.10000000 BTC in unspent supplied transactions
0.06997760 BTC going back to cold storage address 2N93du8YobdgsHyu3qgBvSyhGUT52utMNeA
0.03000000 BTC going to destination address myP4xdJNwAW9iMakvCjnozg814ewgn8apd
Fee amount: 0.00002240

Signing with private keys: 
cQCrT9Ncs9729ao7jbmAWrD9z7tF64s2yKzmD6nkiLAi9sXVZWAn
cP65UeSDZPiTLB6CBwasWv9oJYEjRgQXhswfwcT9HscEKDcEbgy4


Confirm? (y/n): 
Calculating transaction...


Sufficient private keys to execute transaction?
True

Raw signed transaction (hex):
0200000000010153b325ad13ee8dfb31eab188dbfdb8eae8446a6692d7d1bdd56598caa5e8fbed010000002322002065c863c750331cc03efb4daca217e7925cff87f5e9ca9927564234be0d6278e8ffffffff0200c76a000000000017a914ad50fb2489353008e75b0193011ff1f4b76f798c87c0c62d00000000001976a914c3f4800772d58f5429dcb550b1a6a53580175bc188ac04004730440220204946131b118edb9e17a479ecfa3a8ad415286f23c2edfde8b8b196326acaa7022039b22fe4e2855055f6847424da229e93db4bc95f50f7872840e201fdfb6d83290147304402201e36ab7965e55e0a47822d380f78b6950bcf4514a922396c2c8183fe9ecff3ab02201123d7fccba203e1caf0cff9951e84d0139f657a865ab76f5a4e521ae83dffdd018b522103d14ddcfb6817f5579695bbb3eb3e185887bf2942b031e6f716343b8fe7e9e8e221028fcd46f8614b2cbf318096968242a1e22bcfb6d8f2b6dc939c8c27c347b2937b210315acb550120f4cdcb460d5c49080ab508ca4ccd8dedecac22feeac8cd017d4c121022b063ee2f22f9e1982c140fe778672c683111a796dcccdbd47bb65e3d608a98354ae00000000

Transaction fingerprint (md5):
a9b61b8a329f2ef6c7e9a3965dbff095
QR code for transaction written to transaction.png
